app/models/group.rb

```diff
@@ -46,28 +42,32 @@ def submitters
   # Permit a User to receive notification messages for members of this Group
   # @param user [User]
   def enable_messages_for(user:)
-    raise(ArgumentError, "User #{user.uid} is not an administrator for this group #{title}") unless user.can_admin?(self)
-    group_messaging_options << self.class.option_class.new(option_type: self.class.option_class::EMAIL_MESSAGES, group: self, user: user)
+    raise(ArgumentError, "User #{user.uid} is not an administrator or submitter for this group #{title}") unless user.can_admin?(self) || user.can_submit?(self)
+    group = GroupOption.find_or_initialize_by(option_type: GroupOption::EMAIL_MESSAGES, user: user, group: self)
+    group.enabled = true
+    group.save
   end
 
   # Disable a User from receiving notification messages for members of this Group
   # @param user [User]
   def disable_messages_for(user:)
-    raise(ArgumentError, "User #{user.uid} is not an administrator for this group #{title}") unless user.can_admin?(self)
-    users_with_messaging.destroy(user)
+    raise(ArgumentError, "User #{user.uid} is not an administrator or submitter for this group #{title}") unless user.can_admin?(self) || user.can_submit?(self)
+    group = GroupOption.find_or_initialize_by(option_type: GroupOption::EMAIL_MESSAGES, user: user, group: self)
+    group.enabled = false
+    group.save
   end
 
   # Returns true if a given user has notification e-mails enabled for this Group
   # @param user [User]
   # @return [Boolean]
   def messages_enabled_for?(user:)
-    found = users_with_messaging.find_by(id: user.id)
-
-    found.present?
+    group_option = group_messaging_options.find_by(user: user, group: self)
+    group_option ||= GroupOption.new(enabled: true)
+    group_option.enabled
   end
 
   def self.create_defaults
-    return if count > 0
+    return if where(code: "RD").count > 0
```

Comment:
This is a bug that I found when the test creates a group prior to the defaults being created they never get created